src/main/java/com/revature/services/BatchLogicImpl.java

```diff
@@ -0,0 +1,55 @@
+package com.revature.services;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Isolation;
+import org.springframework.transaction.annotation.Transactional;
+
+import com.revature.domain.Batch;
+import com.revature.repositories.BatchRepository;
+
+@Service
+@Transactional(readOnly=false, isolation=Isolation.READ_COMMITTED)
+public class BatchLogicImpl implements BatchLogic {
+
+	@Autowired
+	private BatchRepository dao;
+	
+	@Override
+	public Batch getBatchByName(String batchName) {
+		Batch batch = dao.findByNameIgnoreCase(batchName);
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Immediately return this expression instead of assigning it to the temporary variable "batch". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.revaturelabs.com/coding_rules#rule_key=squid%3AS1488)